### PR TITLE
jinja2 templates in snow DA

### PR DESCRIPTION
**Pull request by aerorahul**
This PR:
- updates the snowDA yamls to jinja2 templates removing all `$( )` and `${ }` template forms.
- all inclusion is now performed via jinja2 `include` with the appropriate indentation.
- jinja2 templates for yamls now carry a `.j2` extension to identify them as templates.
- requires `feature/jinja2_includes` branch from `wxflow`. 

`parse_j2yaml` from `wxflow` now does not render the `$( )` templates.  This change is **not** backwards compatible.

Corressponding PRs in `wxflow` and `global-workflow` are:
- https://github.com/NOAA-EMC/wxflow/pull/15
- https://github.com/NOAA-EMC/global-workflow/pull/2371

### CODE REVIEW

File: parm/snow/hofx/hofx_nomodel.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+state:
+  datapath: '{{ DATA }}/bkg'
+  filetype: fms restart
+  datetime: '{{ current_cycle | to_isotime }}'
+  filename_sfcd: '{{ current_cycle | to_fv3time }}.sfc_data.nc'
+  filename_cplr: '{{ current_cycle | to_fv3time }}.coupler.res'
+  state variables: [snwdph,vtype,slmsk]
+observations:
+{% filter indent(width=2) %}
+{% set snow_obs_list = 'OBS_LIST' | getenv('snow/obs/lists/gdas_snow.yaml.j2') %}
```
Comment (CoryMartin-NOAA) [inline parm/snow/hofx/hofx_nomodel.yaml.j2:33]: I love this!

File: parm/snow/obs/lists/gdas_snow.yaml.j2 [hunk 2]
```diff
@@ -0,0 +1,9 @@
+observers:
+{% filter indent(width=2) %}
+{% include 'snow/obs/config/adpsfc_snow.yaml.j2' %}
+{% include 'snow/obs/config/snocvr_snow.yaml.j2' %}
+{% set cycle = current_cycle | strftime("t%Hz") %}
+{% if cycle in ['t18z'] %}
+{% include 'snow/obs/config/ims_snow.yaml.j2' %}
+{% endif %}
+{% endfilter %}
```
Comment (CoryMartin-NOAA) [inline parm/snow/obs/lists/gdas_snow.yaml.j2]: No EOL here, not sure how big of a deal that is?

**Review by CoryMartin-NOAA — COMMENTED** [review]
Very minor comment, but this looks great, thanks for doing this!

File: parm/snow/obs/lists/gdas_snow.yaml.j2 (hunk 2 above)
Comment (aerorahul) [reply parm/snow/obs/lists/gdas_snow.yaml.j2]: shouldn't be an issue, but added EOL.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**